gguf-py/gguf/gguf_manager.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import annotations
+
+import os
+from enum import IntEnum
+from typing import TYPE_CHECKING, Any, Sequence
+from argparse import Namespace
+from collections import deque
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+from .constants import (
+    GGMLQuantizationType,
+    GGUFEndian,
+    GGUFValueType
+)
+from .gguf_writer import GGUFWriter, WriterState
+
+
+SHARD_NAME_FORMAT = "{:s}-{:05d}-of-{:05d}.gguf"
+METADATA_ONLY_INDICATOR = -1
+
+LLM_KV_SPLIT_NO = "split.no"
+LLM_KV_SPLIT_COUNT = "split.count"
+LLM_KV_SPLIT_TENSORS_COUNT = "split.tensors.count"
+
+KVTempData: TypeAlias = dict[str, tuple[Any, GGUFValueType]] # {key: (value, type)}
+TensorTempData: TypeAlias = tuple[str, np.ndarray[Any, Any], GGMLQuantizationType] # (tensor name, tensor data, tensor dtype)
+Shard: TypeAlias = list[os.PathLike[str], int, int, deque[TensorTempData]] # [shard filename, shard tensor count, shard size, [tensor data]]
+
+
+class SplitStyle(IntEnum):
+    NONE = 0
+    TENSORS = 1
+    SIZE = 2
+
+
+class SplitArguments:
+    def __init__(self, args: Namespace) -> None:
+        self.split = args.split
+        self.split_max_tensors = args.split_max_tensors if args.split else 0
+        self.split_max_size = GGUFManager.split_str_to_n_bytes(args.split_max_size) if args.split and args.split_max_size else 0
+        self.split_style = SplitStyle.NONE if not self.split \
+            else SplitStyle.TENSORS if self.split_max_tensors \
+            else SplitStyle.SIZE
+        self.dry_run = args.dry_run
+        self.small_first_shard = args.small_first_shard
+
+
+class GGUFManager(GGUFWriter):
+    kv_data: KVTempData
+    tensors: list[TensorTempData]
```

Comment:
Let me know what you decide. For now it's not too difficult to just rename the field, since we don't actually care about the superclass `tensors` field - we don't even call the superclass constructor, as we only want the signatures of most of the functions.

Reply:
It turns out we don't even use this field anymore, so I deleted it.